### ISSUE TITLE
Component rendered on the canvas need to removed from the tabindex

### ISSUE THREAD

**Issue by haydn** [haydn/sketchbook-js#71]
# Overview

Elements on the canvas cannot be interacted with using the mouse because pointer events have been been disabled, but they can still be reached using keyboard navigation.

# Steps to reproduce

In macOS/Chrome:

1.  Insert an "Input" component onto the canvas.
2.  Click onto a blank area of the canvas to give it focus and deselect everything.
3.  Press the tab key.

## Expected result

Something in the editor interface gets focus.

## Actual result

The input component gets focus.

**Comment by Mxchaeltrxn** [comment]
I can do this. 

Some questions:
1. What should be selected when pressing tab when nothing is selected? I'm thinking the first element in the layers list.
2. I think an extension of this is that the user should be able to tab/shift-tab to select the next/previous component in the layer list if they're already focused on a single component (nothing should happen onMouseDown). Another possible extension would be to center the component on the screen when tabbing to it. Thoughts on these? It's a bit hard for me to judge the usefulness of these when I haven't really used design apps before.

**Comment by haydn** [comment]
@Mxchaeltrxn Sweet, thanks! I don't think we really need to make keyboard access a priority at the moment (majority of designer will be using a mouse), the main goal is to prevent user accidentally being able to interact with component rendered by the canvas.

To answer your questions:

1.  Whatever is naturally selected if you pressed TAB when there are no input components rendered on the canvas (I assume it's one of the buttons on the left-hand panel, but it's probably not visible because the styles have been reset).
2.  That might worthwhile in the future, but right now it's probably just extra complexity for little user value.

The simplest thing we can probably do is to set `tabindex="-1"` on the inputs when they're rendered, but there might be a way to prevent the whole canvas for being focusable.

**Comment by Mxchaeltrxn** [comment]
@haydn I see two ways to fix this problem. 
1. Add tabIndex="-1" to all tab-able components. This seems not ideal but I'm starting to think it's the only way because of #73 
2. Run a script when focusing on the canvas (or any of it's children) to shift focus to whatever we want to focus through using a ref. However, this means that the input still gets focused and causes #73. 

Doing (1) and adding a comment at the top of the file explaining why tabIndex="-1" is required is what I'd do. We can also add it to the documentation when it starts to get written. Thoughts?

**Comment by haydn** [comment]
@Mxchaeltrxn Given our new architecture an automated version of option 1 seems like it'd be doable. Most of the time developers are going to be importing our pre-made `Canvas` and `Layer` components (in whatever framework/library they're using). I'm thinking we add something to one of those that runs after a component renders and sets all the `tabindex` properties to `-1`.
